src/main/java/de/tum/in/www1/artemis/service/dto/UserPublicInfoDTO.java

```diff
@@ -53,18 +53,10 @@ public UserPublicInfoDTO(User user) {
      * @param userPublicInfoDTO the DTO to assign the roles to
      */
     public static void assignRoleProperties(Course course, User user, UserPublicInfoDTO userPublicInfoDTO) {
-        if (course.getStudentGroupName() != null && user.getGroups().contains(course.getStudentGroupName())) {
-            userPublicInfoDTO.setIsStudent(true);
-        }
-        if (course.getTeachingAssistantGroupName() != null && user.getGroups().contains(course.getTeachingAssistantGroupName())) {
-            userPublicInfoDTO.setIsTeachingAssistant(true);
-        }
-        if (course.getInstructorGroupName() != null && user.getGroups().contains(course.getInstructorGroupName())) {
-            userPublicInfoDTO.setIsInstructor(true);
-        }
-        if (course.getEditorGroupName() != null && user.getGroups().contains(course.getEditorGroupName())) {
-            userPublicInfoDTO.setIsEditor(true);
-        }
+        userPublicInfoDTO.setIsStudent(course.getStudentGroupName() != null && user.getGroups().contains(course.getStudentGroupName()));
+        userPublicInfoDTO.setIsTeachingAssistant(course.getTeachingAssistantGroupName() != null && user.getGroups().contains(course.getTeachingAssistantGroupName()));
+        userPublicInfoDTO.setIsInstructor(course.getInstructorGroupName() != null && user.getGroups().contains(course.getInstructorGroupName()));
+        userPublicInfoDTO.setIsEditor(course.getEditorGroupName() != null && user.getGroups().contains(course.getEditorGroupName()));
```

Comment:
Can the user groups ever contain null? Otherwise we could leave out the null check

Reply:
Removed null check in 4d979930c4a322d63c1401b7707fbad4d055eafb